sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/WriteToDataSourceV2Exec.scala

```diff
@@ -83,12 +85,17 @@ case class CreateTableAsSelectExec(
       catalog.createTable(
         ident, query.schema, partitioning.toArray, properties.asJava) match {
         case table: SupportsWrite =>
-          val batchWrite = table.newWriteBuilder(writeOptions)
+          val writer = table.newWriteBuilder(writeOptions)
             .withInputDataSchema(query.schema)
             .withQueryId(UUID.randomUUID().toString)
-            .buildForBatch()
 
-          doWrite(batchWrite)
+          writer match {
+            case v1: V1WriteBuilder =>
+              val mode = if (ifNotExists) SaveMode.Ignore else SaveMode.ErrorIfExists
```

Comment:
The table was just created above, so both of these modes are incorrect. This should be `SaveMode.Append` because the table already exists.